src/Logger-MCC_DAQ/main.py

```diff
@@ -0,0 +1,196 @@
+                  dictionary.
+                  """
+
+    def __init__(self):
+
+        EmptyDevice.__init__(self)
+
+        self.shortname = "MCC-DAQ"  # short name will be shown in the sequencer
+        self.variables = []
+        self.units = []
+
+        self.port_manager = False
+
+        self.board_num = 0  # only fixed board 0 supported at the moment
+
+        self.data = []  # object to store results before they are returned in 'call'
+
+        self.measurement_modes = {
+            "Single-Ended": "SINGLE_ENDED",
+            "Differential": "DIFFERENTIAL",
+        }
+
+        # AI Range
+        self.available_ai_ranges = {
+            "10 V": ULRange.BIP10VOLTS,
+            "5 V": ULRange.BIP5VOLTS,
+            "2 V": ULRange.BIP2VOLTS,
+            "1 V": ULRange.BIP1VOLTS
+        }
+        self.ai_range = None
+
+    def set_GUIparameter(self):
+
+        gui_parameter = {
+            "Analog input mode": list(self.measurement_modes.keys()),
+            "Analog input channels": "1, 2",
+            "Analog input range": list(self.available_ai_ranges.keys())
+        }
+
+        return gui_parameter
+
+    def get_GUIparameter(self, parameter):
+        """ parse and store GUI options"""
+        self.port_string = parameter["Port"]
+        self.analog_input_mode = self.measurement_modes[parameter["Analog input mode"]]
+        self.analog_inputs_list = parameter["Analog input channels"].split(",")
+        self.analog_inputs = [int(s.strip()) for s in self.analog_inputs_list]
+
+        for i in range(len(self.analog_inputs)):
+            self.variables.append("AI%d" % self.analog_inputs[i])
+            self.units.append("V")
+
+        self.ai_range = self.available_ai_ranges[parameter["Analog input range"]]
+
+    def find_ports(self):
+
+        ul.ignore_instacal()
+        ul.release_daq_device(self.board_num)
+
+        device_list = self.create_device_list()
+
+        if len(device_list) > 0:
+            return device_list
+        else:
+            return ["No device was found"]
+
+    def connect(self):
+
+        ul.ignore_instacal()
+
+        inventory = ul.get_daq_device_inventory(InterfaceType.ANY)
+        device_list = self.create_device_list()
+
+        if self.port_string in device_list:
+            self.descriptor = inventory[device_list.index(self.port_string)]
+        elif self.port_string == "No device was found":
+            msg = "Please use 'Find Ports' and select a valid port to continue."
+            raise Exception(msg)
+        else:
+            msg = "Selected port was not found. Please use 'Find ports' and check whether the port still exists."
+            raise Exception(msg)
+
+        ul.create_daq_device(self.board_num, self.descriptor)
+        ul.flash_led(self.board_num)
+
+    def disconnect(self):
+        ul.release_daq_device(self.board_num)
+
+    def initialize(self):
```

Comment:
All configuration related properties should be moved to semantic standard function "configure" because configure is called whenever a branch of the SweepMe! sequencer starts to be processed. If users have multiple branches with each having a Logger module that is differently configured for the instrument, it would not change anymore because "initialize" is only called once at the beginning.

Therefore, "initialize" only contains commands that need to be used only once at the beginning of each measurement such as the analog input check or the ai range check.

Here, it would make sense to test the driver once being used in two different branches because sometimes there are problems when dlls are involved as they might have exclusive handles.